package/MDAnalysis/auxiliary/base.py

```diff
@@ -0,0 +1,729 @@
+            return self._go_to_step(i)
+        elif isinstance(i, (list, np.ndarray)):
+            return self._list_iter(i)
+        elif isinstance(i, slice):
+            return self._slice_iter(i)
+
+    def _list_iter(self, i):
+        for j in i:
+            yield self._go_to_step(j)
+
+    def _slice_iter(self, i):
+        #TODO - check slice is valid etc
+        for j in range(i.start, i.stop, i.step):
+            yield self._go_to_step(j)
+            
+
+    def _go_to_step(self, i):
+        """ Move to and read i-th auxiliary step. """
+        # Need to define in each auxiliary reader
+        raise NotImplementedError(
+            "BUG: Override _go_to_step() in auxiliary reader!")
+
+    def _reset_frame_data(self):
+        self.frame_data = {}
+
+    def _add_step_to_frame_data(self, ts_time):
+        """ Update ``frame_data`` with values for the current step.
+
+        Parameters
+        ----------
+        ts_time : float
+            the time of the timestep the current step is being 'added to'. Used
+            to calculate difference in time between current step and timestep.
+        """
+        time_diff = self.time - ts_time
+        self.frame_data[time_diff] = self.auxstep.step_data
+
+    def calc_representative(self):
+        """ Calculate represenatative auxiliary value(s) from *frame_data*.
+        
+        Currently available options for calculating represenatative value are:
+          * 'closest': default; the value(s) from the step closest to in time to 
+           the trajectory timestep
+          * 'average': average of the value(s) from steps 'assigned' to the 
+           trajectory timestep.
+        Additionally, if ``cutoff`` is specified, only steps within this time 
+        of the trajectory timestep are considered in calculating the 
+        represenatative.
+
+        If no auxiliary steps were assigned to the timestep, or none fall
+        within the cutoff, representative values are set to ``np.nan``.
+
+        Returns
+        -------
+        Numpy array of auxiliary value(s) 'representing' the timestep.
+        """
+        if self.cutoff == -1:
+            cutoff_data = self.frame_data
+        else:
+            cutoff_data = {key: val for key, val in self.frame_data.items()
+                           if abs(key) <= self.cutoff}
+        if len(cutoff_data) == 0:
+            # no steps are 'assigned' to this trajectory frame, so return
+            # values of ``np.nan``
+            value = self.auxstep._empty_data()
+        elif self.represent_ts_as == 'closest':
+            min_diff = min([abs(i) for i in cutoff_data])
+            # temp fix to get right sign. Go with -ve over +ve.
+            try:
+                value = cutoff_data[-min_diff]
+            except KeyError:
+                value = cutoff_data[min_diff]
+        elif self.represent_ts_as == 'average':
+            value = np.mean(np.array([val for val in cutoff_data.values()]))
```

Comment:
You need to add `axis=0` or you get a single value instead of a value per column.